AndroidAnnotations/androidannotations-core/androidannotations-test/src/main/java/org/androidannotations/test/PageChangeListenerActivity.java

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright (C) 2010-2015 eBusiness Information, Excilys Group
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed To in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.androidannotations.test;
+
+import org.androidannotations.annotations.EActivity;
+import org.androidannotations.annotations.PageScrollStateChanged;
+import org.androidannotations.annotations.PageScrolled;
+import org.androidannotations.annotations.PageSelected;
+
+import android.app.Activity;
+import android.support.v4.view.ViewPager;
+
+@EActivity(R.layout.viewpagers)
+public class PageChangeListenerActivity extends Activity {
+
+	public ViewPager pageScrolledViewPager = null;
+	public int pageScrolledPosition = 0;
+	public float pageScrolledPositionOffset = 0f;
+	public int pageScrolledPositionOffsetPixels = 0;
+
+	public ViewPager pageScrollStateChangedViewPager = null;
+	public int pageScrollStateChangedState = 0;
+
+	public ViewPager pageSelectedViewPager = null;
+	public int pageSelectedPosition = 0;
+
+	@PageScrolled(R.id.viewPager1)
+	public void pageScrolled(ViewPager v, int position, float positionOffset, int positionOffsetPixels) {
+		pageScrolledViewPager = v;
+		pageScrolledPosition = position;
+		pageScrolledPositionOffset = positionOffset;
+		pageScrolledPositionOffsetPixels = positionOffsetPixels;
+	}
+
+	@PageScrollStateChanged(R.id.viewPager1)
+	public void pageScrollStateChanged(ViewPager v, int state) {
+		pageScrollStateChangedViewPager = v;
+		pageScrollStateChangedState = state;
+	}
+
+	@PageSelected(R.id.viewPager1)
+	public void pageSelected(ViewPager v, int position) {
+		pageSelectedViewPager = v;
+		pageSelectedPosition = position;
+	}
```

Comment:
Can you add more test cases (when not all parameters are present)? You can use the same variables because the a new `Activity` is created for each test.